File: ant-sdk/antd-php/src/Models/DataPutPublicResult.php
<?php

declare(strict_types=1);

namespace Autonomi\Antd\Models;

/**
 * Result of a public data put. The DataMap is stored on-network as an extra
 * chunk; `$address` is the shareable retrieval handle.
 */
readonly class DataPutPublicResult
{
    public function __construct(
        /** Hex-encoded on-network DataMap address. */
        public string $address,
        /** Number of chunks stored on the network. */
        public int $chunksStored,
        /** Which payment mode was actually used: "auto", "merkle", or "single". */
        public string $paymentModeUsed,
    ) {}
}

File: ant-sdk/antd-java/src/main/java/com/autonomi/antd/models/DataPutPublicResult.java
package com.autonomi.antd.models;

/**
 * Result of a public data put. The DataMap is stored on-network as an extra
 * chunk; {@code address} is the shareable retrieval handle. Both REST and
 * gRPC populate {@code chunksStored} and {@code paymentModeUsed}.
 *
 * @param address         hex-encoded on-network DataMap address
 * @param chunksStored    number of chunks stored on the network
 * @param paymentModeUsed which payment mode was actually used
 */
public record DataPutPublicResult(
        String address,
        long chunksStored,
        String paymentModeUsed) {
    public DataPutPublicResult(String address) {
        this(address, 0L, "");
    }
}
